tests/unit/models/gpflux/test_models.py

```diff
@@ -0,0 +1,246 @@
+        np.concatenate(
+            [
+                np.random.randn(num_inducing, 2),
+            ],
+            axis=1,
+        )
+    )
+    gp_layer = gpflux.layers.GPLayer(
+        kernel,
+        inducing_variable,
+        num_data=5,
+        num_latent_gps=1,
+        mean_function=gpflow.mean_functions.Zero(),
+    )
+
+    likelihood_layer = gpflux.layers.LikelihoodLayer(gpflow.likelihoods.Gaussian(0.01))
+
+    dgp = DeepGP([keras_layer_1, keras_layer_2, gp_layer], likelihood_layer)
+
+    with pytest.raises(ValueError):
+        DeepGaussianProcess(dgp)
+
+
+def test_dgp_model_attribute(two_layer_model: Callable[[TensorType], DeepGP]) -> None:
+    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
+    dgp = two_layer_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    assert model.model_gpflux is dgp
+
+
+def test_dgp_update(two_layer_model: Callable[[TensorType], DeepGP]) -> None:
+    x = tf.zeros([1, 4])
+    dgp = two_layer_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    assert model.model_gpflux.num_data == 1
+
+    for layer in model.model_gpflux.f_layers:
+        assert layer.num_data == 1
+
+    model.update(Dataset(tf.zeros([5, 4]), tf.zeros([5, 1])))
+
+    assert model.model_gpflux.num_data == 5
+
+    for layer in model.model_gpflux.f_layers:
+        assert layer.num_data == 5
+
+
+@pytest.mark.parametrize(
+    "new_data",
+    [Dataset(tf.zeros([3, 5]), tf.zeros([3, 1])), Dataset(tf.zeros([3, 4]), tf.zeros([3, 2]))],
+)
+def test_dgp_update_raises_for_invalid_shapes(
+    two_layer_model: Callable[[TensorType], DeepGP], new_data: Dataset
+) -> None:
+    x = tf.zeros([1, 4])
+    dgp = two_layer_model(x)
+    model = DeepGaussianProcess(dgp)
+
+    with pytest.raises(ValueError):
+        model.update(new_data)
+
+
+def test_dgp_optimize_with_defaults(two_layer_model: Callable[[TensorType], DeepGP]) -> None:
+    x_observed = np.linspace(0, 100, 100).reshape((-1, 1))
+    y_observed = fnc_2sin_x_over_3(x_observed)
+    data = x_observed, y_observed
+    dataset = Dataset(*data)
+    optimizer = tf.optimizers.Adam()
+    model = DeepGaussianProcess(two_layer_model(x_observed), optimizer=optimizer)
+    elbo = model.model_gpflux.elbo(data)
+    model.optimize(dataset)
+    assert model.model_gpflux.elbo(data) > elbo
+
+
+@pytest.mark.parametrize("batch_size", [10, 100])
+def test_dgp_optimize(two_layer_model: Callable[[TensorType], DeepGP], batch_size: int) -> None:
+    x_observed = np.linspace(0, 100, 100).reshape((-1, 1))
+    y_observed = fnc_2sin_x_over_3(x_observed)
+    data = x_observed, y_observed
+    dataset = Dataset(*data)
+
+    optimizer = tf.optimizers.Adam()
+
+    fit_args = {"batch_size": batch_size, "epochs": 10, "verbose": 0}
+
+    model = DeepGaussianProcess(two_layer_model(x_observed), optimizer, fit_args)
+    elbo = model.model_gpflux.elbo(data)
+    model.optimize(dataset)
+    assert model.model_gpflux.elbo(data) > elbo
```

Comment:
Tests like that make me increasingly uneasy about exposing GPflux in Trieste wrapper. As an optimization toolkit, Trieste shouldn't be verifying internal particularities of GPflux model

Reply:
This is more testing that the `optimize` method (which is a Trieste method) works (to an extent), i.e. that the elbo of the model actually increases as we train. We have similar tests I believe in the GPflow folder. I'm not sure how to otherwise test that the underlying model is actually being changed when we call `optimize`

Reply:
ok, i guess question is a bit out of scope of this PR anyway